# SPDX-License-Identifier: BSD-3-Clause
# Distributed under the OSI-approved BSD 3-Clause License. See LICENSE file for details.
diff --git a/Source/cmake.cxx b/Source/cmake.cxx
index 3b311e0366..c1db5dcb50 100644
--- a/Source/cmake.cxx
+++ b/Source/cmake.cxx
@@ -94,19 +94,18 @@
 #  include "cmVariableWatch.h"
 #endif
 
-#if defined(__MINGW32__) && defined(CMAKE_BOOTSTRAP)
+#if defined(_WIN32) && defined(CMAKE_BOOTSTRAP)
 #  define CMAKE_BOOT_MINGW
 #endif
 
 // include the generator
 #if defined(_WIN32) && !defined(__CYGWIN__)
+#    include "cmGlobalNMakeMakefileGenerator.h"
 #  if !defined(CMAKE_BOOT_MINGW)
 #    include <cmext/memory>
-
 #    include "cmGlobalBorlandMakefileGenerator.h"
 #    include "cmGlobalFastbuildGenerator.h"
 #    include "cmGlobalJOMMakefileGenerator.h"
-#    include "cmGlobalNMakeMakefileGenerator.h"
 #    include "cmGlobalVisualStudio14Generator.h"
 #    include "cmGlobalVisualStudioVersionedGenerator.h"
 #    include "cmVSSetupHelper.h"
@@ -3319,11 +3318,11 @@ void cmake::AddDefaultGenerators()
     cmGlobalVisualStudioVersionedGenerator::NewFactory15());
   this->Generators.push_back(cmGlobalVisualStudio14Generator::NewFactory());
   this->Generators.push_back(cmGlobalBorlandMakefileGenerator::NewFactory());
-  this->Generators.push_back(cmGlobalNMakeMakefileGenerator::NewFactory());
   this->Generators.push_back(cmGlobalJOMMakefileGenerator::NewFactory());
 #  endif
   this->Generators.push_back(cmGlobalMSYSMakefileGenerator::NewFactory());
   this->Generators.push_back(cmGlobalMinGWMakefileGenerator::NewFactory());
+  this->Generators.push_back(cmGlobalNMakeMakefileGenerator::NewFactory());
 #endif
 #if !defined(CMAKE_BOOTSTRAP)
 #  if (defined(__linux__) && !defined(__ANDROID__)) || defined(_WIN32)
diff --git a/Source/kwsys/SystemTools.cxx b/Source/kwsys/SystemTools.cxx
index 7e23c16a95..5a666cd1ae 100644
--- a/Source/kwsys/SystemTools.cxx
+++ b/Source/kwsys/SystemTools.cxx
@@ -72,6 +72,7 @@
 
 #ifdef _MSC_VER
 #  define umask _umask
+typedef unsigned char byte;
 #endif
 
 // support for realpath call
